### PR TITLE
fix: fix issue in s2s policy ids

### PR DESCRIPTION
**Pull request by vburckhardt**
### Description

See https://github.com/terraform-ibm-modules/terraform-ibm-cos/issues/173

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [x] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [x] The PR references a GitHub issue.
- [x] If relevant, a test for the change is included or updated with this PR.
- [x] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: .secrets.baseline [hunk 1]
```diff
@@ -65,7 +65,7 @@
     }
   ],
   "results": {},
-  "version": "0.13.1+ibm.56.dss",
+  "version": "0.13.1+ibm.55.dss",
```
Comment (ocofaigh) [inline .secrets.baseline:68]: Looks like you have an old version of the git submodule locally. This is going backwards. Can you:

1. Ensure that your local primary branch is up to date:
  i. git checkout main
  ii. git pull origin main
  iii. git submodule update --rebase
2. Make sure your dev branch is rebased with remote primary branch:
  i. git checkout fix-s2s
  ii. git pull origin main
3. Run the following command to sync the git submodule with main branch:
  i. git submodule update --rebase

Comment (ocofaigh) [reply .secrets.baseline:68]: PS: The git submodule hook should of caught this (unless your local version pre dates the hook)

File: main.tf [hunk 2]
```diff
@@ -53,14 +53,15 @@ resource "ibm_resource_key" "resource_key" {
 
 locals {
   cos_instance_id      = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].id)[0] : var.existing_cos_instance_id
+  cos_instance_guid    = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].guid)[0] : element(split(":", var.existing_cos_instance_id), length(split(":", var.existing_cos_instance_id)) - 3)
```
Comment (daniel-butler-irl) [inline main.tf:56]: The original was 
```hcl
cos_instance_guid = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].guid)[0] : tolist(data.ibm_resource_instance.cos_instance[*].guid)[0]
```
It looks like you are editing the instance id, to match the guid. Is this a bit fragile? I do like not having the data lookup but which is better/more stable?

File: .secrets.baseline (hunk 1 above)
Comment (vburckhardt) [reply .secrets.baseline]: Yes saw this - usually submodule mix up, should be fixed in later commits

File: main.tf (hunk 2 above)
Comment (vburckhardt) [reply main.tf:56]: It does not take a cos instance id

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**